### PR TITLE
bugfix: emit relative path when relative parameter is set

### LINKED ISSUE #2990: channel.fromPath's `relative` parameter not respected when given a file path
### Expected behavior and actual behavior

When creating a channel using `channel.fromPath()` and providing a path to a specific file, setting the `relative` parameter to `true` doesn't have any effect, so the resulted path is absolute rather than relative.

### Steps to reproduce the problem

Create a `main.nf` file with the following content:
```
data = channel.fromPath("main.nf", relative: true)

process p1 {
    echo true

    input:
    val path from data

    """
    #!/usr/bin/env bash
    echo $path
    """
}
```

### Program output 

```
/absolute/path/to/nextflow/script/main.nf
```

### Environment 

* Nextflow version: 21.10.6
* Java version: 1.8.0_321
* Operating system: macOS
* Bash version: GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin20)

### Additional context

Expected output is:
```
main.nf
```

Adding an asterisk at the end of the path (i.e. `data = channel.fromPath("main.nf*", relative: true)` solves it, but isn't really a good workaround.

### PR DESCRIPTION
**Pull request by jorgeaguileraseqera**
closes #2990

### CODE REVIEW

**Comment by pditommaso** [comment]
and there's any test? 😄

**Comment by jorgeaguileraseqera** [comment]
> and there's any test? smile

of course but forget to commit it

File: modules/nf-commons/src/main/nextflow/file/FileHelper.groovy [hunk 1]
```diff
@@ -1038,10 +1038,10 @@ class FileHelper {
         final checkIfExists = opts?.checkIfExists as boolean
         final followLinks = opts?.followLinks == false ? [LinkOption.NOFOLLOW_LINKS] : Collections.emptyList()
         if( !checkIfExists || FilesEx.exists(result, followLinks as LinkOption[]) ) {
-            return result
+            return opts?.relative ? path : result
         }
 
-        throw new NoSuchFileException(FilesEx.toUriString(result))
+        throw new NoSuchFileException(FilesEx.toUriString(opts?.relative ? path : result))
```
Comment (pditommaso) [inline modules/nf-commons/src/main/nextflow/file/FileHelper.groovy:1044]:
```suggestion
        throw new NoSuchFileException(opts?.relative ? path : FilesEx.toUriString(result))
```

I think when it's relative makes no sense to use `FilesEx.toUriString`

File: modules/nextflow/src/test/groovy/nextflow/ChannelTest.groovy [hunk 2]
```diff
@@ -269,6 +269,28 @@ class ChannelTest extends Specification {
...
+        then:
+        result*.toString() == [ file1.absolutePath ]
+
+        when:
+        result = Channel
+                .fromPath([relative:true], 'alpha.txt')
+                .toSortedList().getVal()
+        then:
+        result*.toString() == [ 'alpha.txt' ]
+
```
Comment (pditommaso) [inline modules/nextflow/src/test/groovy/nextflow/ChannelTest.groovy:290]: Let's add a test for the default case (absence of `relative` option)

**Pull request merged**